### PR TITLE
GH-34080: [Python] Add support for round_binary to python

### LINKED ISSUE #34080: [Python] Remove warning "Python binding for RoundBinaryOptions not exposed"
### Describe the bug, including details regarding any error messages, version, and platform.

This warning has been showing up since the addition of the round_binary kernel.  I assume we simply need to add bindings.

### Component(s)

Python

### PR DESCRIPTION
**Pull request by westonpace**
* Closes: #34080

### CODE REVIEW

File: python/pyarrow/tests/test_array.py [hunk 1]
```diff
@@ -714,7 +714,8 @@ def test_struct_from_arrays():
         pa.StructArray.from_arrays(
             arrays, fields, mask=pa.chunked_array([mask]))
 
-    # Non-empty array with no fields https://github.com/apache/arrow/issues/15109
```
Comment (Fokko) [inline python/pyarrow/tests/test_array.py:717]: Nit: Looks like an unnecessary change

**Review by Fokko — APPROVED** [review]

File: python/pyarrow/tests/test_array.py (hunk 1 above)
Comment (westonpace) [reply python/pyarrow/tests/test_array.py:717]: pylint was complaining for me.  But, oddly, it was complaining about an 80 character line limit and I thought we allowed 90.  I'll revert and see if CI passes.  Maybe I have a stray .pylintrc in my environment somewhere.

Comment (westonpace) [reply python/pyarrow/tests/test_array.py]: Ah, my install of archery was out of date it seems.  I updated it and the warning went away.  I suspect we changed where we store the flake config.

Comment (westonpace) [reply python/pyarrow/tests/test_array.py]: Thanks for the catch

Comment (Fokko) [reply python/pyarrow/tests/test_array.py]: I believe PyArrow allows 87 chars at most. Thanks for fixing 👍🏻

**Review by wjones127 — APPROVED** [review]
